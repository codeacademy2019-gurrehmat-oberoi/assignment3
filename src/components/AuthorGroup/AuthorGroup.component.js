import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Book from '../Book/Book.component';
import './AuthorGroup.css';

class AuthorGroup extends Component {
  render() {
    const { booksArray, author } = this.props;
    const books = booksArray.map((book, index) => (
      <Book bookData={book} key={index} />
    ));
    return (
      <div className="authorGroup">
        <div className="author">{author}</div>
        <div className="books">
          {books}
        </div>
      </div>
    );
  }
}

AuthorGroup.defaultProps = {
  booksArray: [
    {
      title: 'Harry Potter and The Sorcerers Stone',
      rating: 4.5,
    },
    {
      title: 'Harry Potter and The Sorcerers Stone',
      rating: 4.5,
    },
    {
      title: 'Harry Potter and The Sorcerers Stone',
      rating: 4.5,
    },
  ],
  author: 'JK Rowling',
};

AuthorGroup.propTypes = {
  booksArray: PropTypes.array,
  author: PropTypes.string,
};

export default AuthorGroup;
