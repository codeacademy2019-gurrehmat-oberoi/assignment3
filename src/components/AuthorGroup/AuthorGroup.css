.authorGroup {
  border: 3px solid;
  border-radius: 30px;
  display: flex;
  flex-direction: column;
  align-content: center;
  margin: 20px;
  margin-left: 20vw;
  margin-right: 20vw;
}

.authorGroup .author {
  align-self: center;
  font-size: 30px;
  border: 2px solid;
  border-radius: 20px;
  padding: 10px;
  margin: 10px;
}

.authorGroup .books{
  display: flex;
  align-self: center;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
}