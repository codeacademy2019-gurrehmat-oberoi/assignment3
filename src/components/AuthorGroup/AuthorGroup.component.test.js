import renderer from 'react-test-renderer';
import React from 'react';
import AuthorGroup from './AuthorGroup.component';

describe('Author Group', () => {
  it('should render an author group', () => {
    const tree = renderer.create(<AuthorGroup />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
