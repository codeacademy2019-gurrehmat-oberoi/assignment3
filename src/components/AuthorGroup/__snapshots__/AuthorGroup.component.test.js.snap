// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Group should render an author group 1`] = `
<div
  className="authorGroup"
>
  <div
    className="author"
  >
    JK Rowling
  </div>
  <div
    className="books"
  >
    <div
      className="book"
    >
      <img
        alt="Book cover"
        className="bookCover"
        src="cover_image.jpeg"
      />
      <div
        className="bookTitle"
      >
        Harry Potter and The Sorcerers Stone
      </div>
      <div
        className="bookAppreciation"
      >
        <div
          className="bookRating"
        >
          4.5
        </div>
        <div
          className="bookLikeButton"
        >
          <img
            alt="like book"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="book"
    >
      <img
        alt="Book cover"
        className="bookCover"
        src="cover_image.jpeg"
      />
      <div
        className="bookTitle"
      >
        Harry Potter and The Sorcerers Stone
      </div>
      <div
        className="bookAppreciation"
      >
        <div
          className="bookRating"
        >
          4.5
        </div>
        <div
          className="bookLikeButton"
        >
          <img
            alt="like book"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="book"
    >
      <img
        alt="Book cover"
        className="bookCover"
        src="cover_image.jpeg"
      />
      <div
        className="bookTitle"
      >
        Harry Potter and The Sorcerers Stone
      </div>
      <div
        className="bookAppreciation"
      >
        <div
          className="bookRating"
        >
          4.5
        </div>
        <div
          className="bookLikeButton"
        >
          <img
            alt="like book"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-red.svg",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
